packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -48,11 +67,26 @@ export function cloudflare<T extends Record<string, WorkerOptions>>(
 		},
 		configResolved(resolvedConfig) {
 			viteConfig = resolvedConfig;
+			normalizedPluginConfig = normalizePluginConfig(
+				pluginConfig,
+				resolvedConfig,
+			);
+		},
+		resolveId(source) {
+			const worker = normalizedPluginConfig.workers[this.environment.name];
+			if (worker) {
+				return resolveNodeAliases(source, worker.workerOptions);
+			}
+		},
+		transform(code, id) {
+			const worker = normalizedPluginConfig.workers[this.environment.name];
+			if (worker) {
+				if (id === worker.entryPath) {
```

Comment:
Does this work? I would have thought that `id` is absolute and `worker.entryPath` is relative but I might be missing something.

Reply:
Also, I'm thinking now that my suggestion of using `path.resolve` may not be quite right because we should also support package imports for `main`. In that case we'd need `import.meta.resolve` or some other solution but I'm happy to keep that as a todo for now.

Reply:
> Does this work? I would have thought that id is absolute and worker.entryPath is relative but I might be missing something.

I think you are right. This seems to pass the tests but perhaps that is luck as much as anything else.
`entryPoint` is literally `main` from the plugin config, so it will be as relative as that is...

Reply:
Nice solution! I forgot about `this.resolve`.